pkg/reconciler/v1alpha1/route/route.go

```diff
@@ -413,8 +423,57 @@ func objectRef(a accessor, gvk schema.GroupVersionKind) corev1.ObjectReference {
 	}
 }
 
-func routeDomain(ctx context.Context, route *v1alpha1.Route) string {
+// Save these as global to save time on each call to routeDomain()
+var savedTemplate *template.Template = nil
+var savedText string = ""
+
+func routeDomain(ctx context.Context, route *v1alpha1.Route) (string, error) {
 	domainConfig := config.FromContext(ctx).Domain
 	domain := domainConfig.LookupDomainForLabels(route.ObjectMeta.Labels)
-	return fmt.Sprintf("%s.%s.%s", route.Name, route.Namespace, domain)
+
+	// These are the available properties they can choose from.
+	// We could add more over time - e.g. RevisionName if we thought that
+	// might be of interest to people.
+	data := struct {
```

Comment:
Should the be a proper type vs. an anonymous one?

Reply:
I'm not a golang expert, does it matter if it's only used in this one func?

Reply:
It does not and it's not a golang thing. It's more a comment on code structure. If you have a defined type for this, you can just point at it in documentation and say: all fields of the type XYZ can be used in the gotemplate.

You can also reuse that type in tests potentially.

Reply:
Good reasons - done.  BTW, I switched "DomainTemplate" back to "RouteTemplate" :-)  I forgot that the template isn't defining the "domain", it's defining the entire "route" of the service. And I didn't want people to get confused and think that they're just defining the domain name instead of the entire URL's hostname value.